sympy/integrals/integrals.py

```diff
@@ -94,6 +94,29 @@ def __new__(cls, function, *symbols, **assumptions):
     def __getnewargs__(self):
         return (self.function,) + tuple([tuple(xab) for xab in self.limits])
 
+    # fixes issue #20809
+    def has_free(obj, syms):
```

Comment:
could the signature be like `has`: `has(self, *patterns)` -- star vs iterable?

Reply:
To be efficient syms should be a set. Using `*syms` would force a double conversion from set to tuple and back since you'd have
```
def has_free_in(obj, *syms):
    syms = set(syms)
    ...

syms = {a, b, c}
has_free_in(obj, *syms)
```

Reply:
Can we get a test for 
```python
if not iterable(syms): raise TypeError('`syms` should be an iterable container of objects')
```

Reply:
Sure, `Integral(x, (x, 0, 1)).has_free(x)` will raise `TypeError`.
Thanks.